src/main/java/org/apache/cassandra/sidecar/restore/RestoreJobManager.java

```diff
@@ -195,17 +195,13 @@ private Future<Void> deleteDataOfJobAsync(UUID jobId)
             {
                 LOGGER.warn("Error on listing staged restore job directories. Path={}", stagingDir, ioe);
             }
-            finally
-            {
-                promise.complete(); // always completes the promise regardless of the deletion result
```

Comment:
is this guaranteed now?

Reply:
it is now guaranteed by the `executeBlocking(Callable)` api.